File: backend/app/schemas/export.py
"""Pydantic schemas for export API responses."""

from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class ExportVersionSchema(BaseModel):
    """Schema for a single versioned export artifact."""

    filename: str
    doc_type: str  # FDS or SDS
    mode: str  # draft or final
    language: str  # nl or en
    version: str
    created_at: datetime
    size_bytes: int
    download_url: str


class ExportVersionListResponse(BaseModel):
    """Response schema for listing export versions."""

    project_id: int
    versions: list[ExportVersionSchema]


class AssemblyReadinessSchema(BaseModel):
    """Schema for assembly readiness check response."""

    ready: bool
    mode: str
    unreviewed_phases: list[str]
    has_content: bool


class PandocStatusSchema(BaseModel):
    """Schema for Pandoc installation status."""

    installed: bool
    version: Optional[str] = None


class ExportProgressEvent(BaseModel):
    """Schema for SSE progress event data."""

    event: str  # step_start | step_done | complete | error | cancelled
    step: int
    name: str
    total_steps: int
    message: Optional[str] = None

File: frontend/src/types/export.ts
export interface ExportVersion {
  filename: string
  doc_type: 'FDS' | 'SDS'
  mode: 'draft' | 'final'
  language: 'nl' | 'en'
  version: string
  created_at: string
  size_bytes: number
  download_url: string
}

export interface ExportVersionList {
  project_id: number
  versions: ExportVersion[]
}

export interface AssemblyReadiness {
  ready: boolean
  mode: string
  unreviewed_phases: string[]
  has_content: boolean
}

export interface PandocStatus {
  installed: boolean
  version: string | null
}

export type PipelineStageStatus = 'idle' | 'running' | 'done' | 'error'

export interface PipelineStage {
  step: number
  name: string
  status: PipelineStageStatus
  errorMessage?: string
}

export interface ExportProgressEvent {
  event: 'step_start' | 'step_done' | 'complete' | 'error' | 'cancelled'
  step: number
  name: string
  total_steps: number
  message?: string
  artifact_filename?: string
}
